src/H5Tconv.c

```diff
@@ -4221,6 +4303,49 @@ H5T__conv_ref(const H5T_t *src, const H5T_t *dst, H5T_cdata_t *cdata,
     } /* end switch */
 
 done:
+    /* Release converted elements on error */
+    if (ret_value < 0 && conversions_made) {
```

Comment:
After looking at this and the above side by side, it looks like the only difference is whether we call `H5T_vlen_reclaim_elmt` or `H5Rdestroy` on the element. Would be much nicer inside a helper function.

Reply:
Agree, but I'll leave this for later work. Right now I just want to get this in.